server/embedded/src/org/labkey/embedded/LabKeyServer.java

```diff
@@ -309,46 +320,33 @@ private List<ContextResource> getDataSourceResources(ContextProperties props, St
                 return dataSourceResources;
             }
 
-            private List<ContextResource> getExtraContextResources(ContextProperties contextProperties) throws ConfigException
+            private void addExtraContextResources(ContextProperties contextProperties, StandardContext context) throws ConfigException
             {
-                List<ContextResource> contextResources = new ArrayList<>();
-                Map<String, Map<String, String>> resourceMaps = Objects.requireNonNullElse(contextProperties.getResources(), Collections.emptyMap());
+                Map<String, Map<String, Map<String, String>>> resourceMaps = Objects.requireNonNullElse(contextProperties.getResources(), Collections.emptyMap());
 
-                for (Map.Entry<String, Map<String, String>> entry : resourceMaps.entrySet())
+                for (Map.Entry<String, Map<String, Map<String, String>>> parentEntry : resourceMaps.entrySet())
                 {
-                    Map<String, String> resourceMap = new CaseInsensitiveKeyMap<>();
-                    resourceMap.putAll(entry.getValue());
-                    if (!resourceMap.containsKey("name"))
+                    for (Map.Entry<String, Map<String, String>> entry : parentEntry.getValue().entrySet())
                     {
-                        throw new ConfigException("Resource configuration error: Unnamed resource found 'context.resources.%s'".formatted(entry.getKey()));
-                    }
-                    if (!resourceMap.containsKey("type"))
-                    {
-                        throw new ConfigException("Resource configuration error: 'type' is not defined for resource '%s'".formatted(resourceMap.get("name")));
-                    }
+                        Map<String, String> resourceMap = new CaseInsensitiveKeyMap<>();
+                        resourceMap.putAll(entry.getValue());
+                        String resourceTypeString = parentEntry.getKey();
 
-                    ContextResource contextResource = new ContextResource();
-                    // Handle resource properties with explicit setters
-                    contextResource.setName(resourceMap.remove("name"));
-                    contextResource.setType(resourceMap.remove("type"));
-                    contextResource.setDescription(resourceMap.remove("description"));
-                    contextResource.setLookupName(resourceMap.remove("lookupName"));
-                    if (resourceMap.containsKey("scope"))
-                    {
-                        contextResource.setScope(resourceMap.remove("scope"));
-                    }
-                    contextResource.setAuth(Objects.requireNonNullElse(resourceMap.remove("auth"), "Container"));
+                        ResourceType type;
```

Comment:
Just for clarity, maybe a different variable name for this `ResourceType`. "type" is one of the resource parameters, so It's a bit confusing to have "type" mean something different here.